File: src/main/java/com/pocketempire/units/UnitStats.java
package com.pocketempire.units;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Map;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UnitStats {
    private String type;
    private int hp;
    private int attack;
    private int defense;
    private int movement;
    private int range;
    private int cost;
    private MovementType movementType;
    @com.fasterxml.jackson.annotation.JsonProperty("role") private UnitRole unitRole;
    private double effectChance;
    private String requiredTech;
    private Map<String, UnitRole> roleByAbility;
}

File: src/main/kotlin/com/pocketempire/units/UnitStats.kt
package com.pocketempire.units

import com.fasterxml.jackson.annotation.JsonProperty

data class UnitStats(
    var type: String = "",
    var hp: Int = 0,
    var attack: Int = 0,
    var defense: Int = 0,
    var movement: Int = 0,
    var range: Int = 0,
    var cost: Int = 0,
    var movementType: MovementType? = null,

    @field:JsonProperty("role")
    var unitRole: UnitRole? = null,

    var effectChance: Double = 0.0,
    var requiredTech: String? = null,
    var roleByAbility: Map<String, UnitRole> = emptyMap()
)
